%\VignetteIndexEntry{TSCAN: Tools for Single-Cell ANalysis} 
%\VignetteDepends{} 
%\VignettePackage{TSCAN}

\documentclass[10pt,oneside]{article}
\newcommand{\thetitle}{TSCAN: Tools for Single-Cell ANalysis}

\title{\textsf{\textbf{\thetitle}}}
\author{Zhicheng Ji\\[1em]Johns Hopkins University,\\ Baltimore, Maryland, USA\\
\texttt{zji4@jhu.edu} \and
Hongkai Ji\\[1em]Johns Hopkins University,\\ Baltimore, Maryland, USA\\
\texttt{hji@jhsph.edu}}

\begin{document}

\maketitle

\tableofcontents

\section{Introductions}

TSCAN currently provides a bunch of essential tools to perform analysis on single-cell RNA-seq data. Potentially TSCAN can also be used to analyze other kinds of data generated by single-cell experiments and with similar data formats.

The most important function provided by TSCAN is to construct a pseudotemporal ordering and time course for single cells. TSCAN first uses principal component analysis to perform dimension reduction on the preprocessed gene expression data. Then the travelling salesman problem (TSP) algorithm is applied to construct a pseudotemporal ordering of cells so that the total distance along the path is nearly the shortest one. To address the problem that TSP does not have a designated starting/ending point, users can provide prior biological information of gene expression to select the optimal starting point of the ordering. After the pseudotemporal ordering and pseudotime are constructed, TSCAN can search for differentially expressed genes.

TSCAN comes with a user-friendly GUI written in shiny which provides powerful and convenient functions to tune pseudotemporal ordering. For example, users can trim unwanted cells or branches after an initial construction of pseudotime ordering. Users can also use multiple marker genes to tune the starting point of the TSP path.

\section{Preprocess Gene Expression Profiles}

The function preprocess can be used to filter out genes which are not expressed in most cells or not differentially expressed. 

<<>>=
library(TSCAN)
library(HSMMSingleCell)
library(Biobase)
data(HSMM)
HSMMdata <- exprs(HSMM)
HSMMdata <- HSMMdata[,grep("T0|72",colnames(HSMMdata))]
procdata <- preprocess(HSMMdata)
str(procdata)
@

By default, preprocess will first take log2 on the original dataset after adding a pseudocount of 1. Then the function will rule out genes which have expression values of less than 1 in at least half of all cells. Genes whose coefficient of covariance of the expression values is less than 1 are also filtered out.

\section{Constructing Pseudotime}

The function TSPpseudotime will perform dimension reduction using principal component analysis and construct a pseudotemporal ordering using TSP algorithm. Because there is no designated starting point of TSP path, it is highly recommended that users specify prior biological information of gene expression to obtain optimal starting point. Here the expression of gene MYOG is used as prior biological information. In this example the reduced dimension is set to be 2 for better visualization results. If no dimension is specified, TSPpseudotime will automatically choose the optimal dimensionality.

<<>>=
MYOGexpr <- log2(HSMMdata["ENSG00000122180.4",]+1)
HSMMpseudotime <- TSPpseudotime(procdata, geneexpr = MYOGexpr, dim = 2)
@

Use plotpseudotime function to visualize the constructed TSP based on the data after dimension reduction. The MYOG expression values are used to define the size of the nodes in the plot.

<<>>=
plotpseudotime(HSMMpseudotime, markerexpr = MYOGexpr)
@

\section{Testing Differentially Expressed Genes}

Use difftest function to detect differentially expressed genes given a constructed pseudotemporal ordering. The function will compare fitting a generalized additive model (gam) and fitting a constant. If gam fits significantly better than a constant fit than the gene is supposed to be differentially expressed. Original p-values as well as q-values after adjusting for multiple testing will be the output.

<<>>=
diffval <- difftest(procdata,HSMMpseudotime[[1]])
#Selected differentially expressed genes under qvlue cutoff of 0.05
row.names(diffval)[diffval$qval < 0.05]
@

Use singlegeneplot function to plot the expression value of a single gene against a given pseudotime.
<<>>=
singlegeneplot(MYOGexpr, HSMMpseudotime[[1]])
@

\section{Comparing Different Pseudotemporal Ordering}

Given sub-population information of the single-cell information, function orderscore can calculate the pseudotemporal ordering score (POS) of multiple pseudotemporl ordering.

<<>>=

subpopulation <- data.frame(cell = colnames(procdata), sub = ifelse(grepl("T0",colnames(procdata)),0,1), stringsAsFactors = FALSE)
#Comparing ordering with or without marker gene information
order1 <- TSPpseudotime(procdata, geneexpr = MYOGexpr, dim = 2)
order2 <- TSPpseudotime(procdata, dim = 2)
orders <- list(order1[[1]],order2[[1]])
orderscore(subpopulation, orders)
@

From the previous example we can see that when prior information of MYOG gene expression is available, POS will dramatically increase compared to the ordering without MYOG gene expression information.

\end{document}

